app/controllers/users_controller.rb

```diff
@@ -0,0 +1,23 @@
+class UsersController < ApplicationController
+  def user_data
+    @user_data = User.select('name, COUNT(posts.id) AS post_count')
+      .joins(:posts)
+      .group('users.id')
+  end
+
+  def show
+    @user = User.find(params[:id])
+    @user_posts = @user.posts
+  end
+
+  def show_posts
+    @user = User.find(params[:user_id])
+    @user_posts = @user.posts.includes(comments: :user)
+  end
+
+  def show_post
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:post_id])
+    @post_comments = @post.comments.includes(:user)
+  end
```

Comment:
- [ ] You're meant to have two controllers, `users_controller` and `posts_controller` with #index and #show actions. as listed in your routes from the requirements. 
- Kindly create a post controller with the actions
- Lastly, create a view page for the controller also 👍 
```
    # app/views/posts/index.html.erb
    # app/views/posts/show.html.erb
```

### Sample:
```
# app/controllers/posts_controller.rb
class PostsController < ApplicationController
  
  def index; end

  def show; end

end
```